src/mudlet.cpp

```diff
@@ -2341,16 +2352,15 @@ void mudlet::slot_notes()
 
 void mudlet::slot_irc()
 {
-    if( ! mpIRC )
+    Host * pHost = getActiveHost();
+    if( ! pHost ) return;
+
+    if( ! mpIrcClientMap[pHost] )
     {
-        mpIRC = new dlgIRC();
-        mpIRC->setWindowTitle( tr( "Mudlet live IRC Help Channel #mudlet-help on irc.freenode.net" ) );
-        mpIRC->setWindowIcon( QIcon( QStringLiteral( ":/icons/mudlet_irc.png" ) ) );
```

Comment:
You still want the icon I think (unless it is set elsewhere) and, given that there could be multiple IRC dialogues floating around it would be a good idea to entitle them with:

    mpIRC->setWindowTitle( tr( "Mudlet IRC Chat - %1 - %2 on %3" ).arg(pHost-getName(), ?Nickname?, ?Server? );

With ?Nickname? and ?Server? filled in...